common/gce.py

```diff
@@ -54,3 +54,55 @@ def get_preempted_instances(project, zone):
         if (instance['scheduling']['preemptible'] and
                 instance['status'] == 'TERMINATED'):
             yield instance['name']
+
+
+def get_instance_group_managers():
+    """Returns the instance group managers resource."""
+    return thread_local.service.instanceGroupManagers()
+
+
+def get_instance_group_size(instance_group: str, project: str,
+                            zone: str) -> int:
+    """Returns the number of instances running in |instance_group|."""
+    managers = get_instance_group_managers()
+    request = managers.get(instanceGroupManager=instance_group,
+                           project=project,
+                           zone=zone)
+    return request.execute()['targetSize']
+
+
+def resize_instance_group(size, instance_group, project, zone):
+    """Changes the number of instances running in |instance_group| to |size|."""
+    assert size >= 1
+    managers = get_instance_group_managers()
+    request = managers.resize(instanceGroupManager=instance_group,
+                              size=size,
+                              project=project,
+                              zone=zone)
+    return request.execute()
+
+
+def delete_instance_group(instance_group, project, zone):
+    """Deletes |instance_group|."""
+    managers = get_instance_group_managers()
+    request = managers.delete(instanceGroupManager=instance_group,
+                              zone=zone,
+                              project=project)
+    return request.execute()
+
+
+def create_instance_group(name: str, instance_template_url: str,
+                          base_instance_name: str, project: str, zone: str):
+    """Creates an instance group named |name| from the template specified by
+    |instance_template_url|."""
+    managers = get_instance_group_managers()
+    target_size = 1
```

Comment:
probably should add size as an argument to this function and can default=1
size=1

Reply:
Don't think this is necessary.